test/BIP39Wallet.Tests.cs

```diff
@@ -1,109 +1,98 @@
-                accountInfo.PrivateKey.PublicKey.Decompress()
-            );
-            Assert.Equal("2ihA5K7sSsA78gekyhuh7gcnX4JkGVqJmSGnf8Kj1hZefR4sX5", accountInfo.PrivateKey.PublicKey.Decompress().ToAddress());
         }
-
-        [Fact]
-        public void GetWalletByPrivateKey_ReturnsValidAccountInfo()
+        else
         {
-            var privateKey = "f0c3bf2cfc4f50405afb2f1236d653cf0581f4caedf4f1e0b49480c840659ba9";
-            var accountInfo = PrivateKey.Parse(privateKey);
-            var keyTest =
-                new PublicKey(
-                    "04c0f6abf0e3122f4a49646d67bacf85c80ad726ca781ccba572033a31162f22e55a4a106760cbf1306f26c25aea1e4bb71ee66cb3c5104245d6040cce64546cc7");
-            Assert.NotNull(accountInfo);
-            Assert.Equal(
-                keyTest,
-                accountInfo.PublicKey.Decompress()
-            );
-
-            var compareResult = accountInfo.PublicKey.CompareTo(keyTest);
-            Assert.Equal(0, compareResult);
-
-            var keyHash = accountInfo.PublicKey.GetHashCode();
-            Assert.Equal(1373039134, keyHash);
-            Assert.Equal("04c0f6abf0e3122f4a49646d67bacf85c80ad726ca781ccba572033a31162f22e55a4a106760cbf1306f26c25aea1e4bb71ee66cb3c5104245d6040cce64546cc7", accountInfo.PublicKey.ToString());
-            Assert.Equal("2ihA5K7sSsA78gekyhuh7gcnX4JkGVqJmSGnf8Kj1hZefR4sX5", accountInfo.PublicKey.Decompress().ToAddress());
+            Assert.Throws<ArgumentException>(() => new AElfWalletFactory().Create(language: language));            
         }
+    }
 
-        [Fact]
-        public void Sign_ReturnsValidSignature()
-        {
-            const string privateKey = "03bd0cea9730bcfc8045248fd7f4841ea19315995c44801a3dfede0ca872f808";
-            const string hash = "68656c6c6f20776f726c643939482801";
-            const string signed =
-                "59EF1D3B2B853FCA1E33D07765DEBAAF38A81442CFE90822D4334E8FCE9889D80C99A0BE1858C1F26B4D99987EFF6003F33B7C3F32BBDB9CEEC68A1E8A4DB4B000";
-
-
-            // Arrange
-            var result = PrivateKey.Parse(privateKey).Sign(Encoding.UTF8.GetBytes(hash));
-            var hex = new StringBuilder(result.Length * 2);
-            foreach (var b in result)
-            {
-                hex.AppendFormat("{0:x2}", b);
-            }
+    [Fact]
+    public void GetWalletByMnemonic_ReturnsValidAccountInfo()
+    {
+        // Arrange
+        const string mnemonic = "put draft unhappy diary arctic sponsor alien awesome adjust bubble maid brave";
+        var accountInfo = new AElfWalletFactory().FromMnemonic(mnemonic).Derive(0);
+        Assert.NotNull(accountInfo);
+        Assert.Equal(
+            new PublicKey(
+                PubKey),
+            accountInfo.PrivateKey.PublicKey.Decompress()
+        );
+        Assert.Equal("2ihA5K7sSsA78gekyhuh7gcnX4JkGVqJmSGnf8Kj1hZefR4sX5", accountInfo.PrivateKey.PublicKey.Decompress().ToAddress());
```

Comment:
maybe "2ihA5K7sSsA78gekyhuh7gcnX4JkGVqJmSGnf8Kj1hZefR4sX5" should be extracted out as a const (im guessing its the wallet-address?) and referenced here insted?

Reply:
done